compact.go

```diff
@@ -145,6 +146,22 @@ func (c *compactor) Plan() ([][]string, error) {
 		return [][]string{res}
 	}
 
+	// Check if any "large" blocks need to be recompacted due to deletions.
+	var compactDirs [][]string
+	for _, dm := range dms {
+		if dm.meta.MaxTime-dm.meta.MinTime > int64(c.opts.maxBlockRange/2) {
+			if dm.meta.Stats.NumTombstones > 0 &&
+				(dm.meta.Stats.NumSeries/dm.meta.Stats.NumTombstones) <= 20 { // 5%
+				compactDirs = append(compactDirs, []string{dm.dir})
+			}
+			continue
+		}
```

Comment:
Better to reduce levels of indentation by inverting the first if and putting the `break` inside of it.